src/main/java/net/mcreator/ui/modgui/BlockGUI.java

```diff
@@ -320,6 +320,9 @@ public BlockGUI(MCreator mcreator, ModElement modElement, boolean editingMode) {
 				Dependency.fromString(
 						"x:number/y:number/z:number/world:world/entity:entity/direction:direction/blockstate:blockstate/hitX:number/hitY:number/hitZ:number"));
 
+		specialInformation = new TextProcedureSelector(null, mcreator, new JTextField(25),
+				Dependency.fromString("itemstack:itemstack/world:world"));

```

Comment:
ItemStacks have `getAttachedEntity()` method in 1.16.5 version (`getEntityRepresentation()` for 1.18.2), you can use it to also provide entity and coordinates here. The method returns the entity holding given item or an ItemEntity if e. g. an entity threw it away (afair), but it can also return null if item is e. g. inside a chest, you need to note that in the corresponding help tip.